File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities.TestPlugin/UiRibbon/ButtonCreator.vb
Imports Autodesk.Windows
Imports cadwiki.DllReloader.AutoCAD
Imports cadwiki.DllReloader.AutoCAD.UiRibbon.Buttons
Imports System.Drawing
Imports System.Reflection

Namespace cadwiki.AutoCAD2021.Base.Utilities.TestPlugin.UiRibbon
    Friend Class ButtonCreator
        Public Shared Function Create(ByVal name As String,
                                      ByVal text As String,
                                      ByVal tooltip As String,
                                      ByVal bitMap As Bitmap,
                                      ByVal assemblyName As String,
                                      ByVal fullClassName As String,
                                      ByVal methodName As String,
                                      ByVal parameters As Object(),
                                      ByVal appDomainReloader As AutoCADAppDomainDllReloader,
                                      ByVal iExtensionAppAssembly As Assembly) As RibbonButton


            Dim ribbonButton = New RibbonButton()
            ribbonButton.Name = name
            ribbonButton.ShowText = True
            ribbonButton.Text = text
            ribbonButton.Size = RibbonItemSize.Standard

            If bitMap IsNot Nothing Then
                Dim image = FileStore.Bitmaps.CreateBitmapSourceFromGdiBitmapForAutoCADButtonIcon(bitMap)
                If image IsNot Nothing Then
                    ribbonButton.Image = image
                    ribbonButton.ShowImage = True
                End If
            End If

            Dim uiRouter = New UiRouter(assemblyName, fullClassName, methodName, parameters, appDomainReloader, iExtensionAppAssembly)
            ribbonButton.CommandParameter = uiRouter
            ribbonButton.CommandHandler = New GenericClickCommandHandler()
            ribbonButton.ToolTip = tooltip
            Return ribbonButton
        End Function
    End Class
End Namespace

File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities.TestPlugin/UiRibbon/ButtonCreator.cs
using System.Drawing;
using System.Reflection;
using Autodesk.Windows;
using cadwiki.DllReloader.AutoCAD;
using cadwiki.DllReloader.AutoCAD.UiRibbon.Buttons;

namespace cadwiki.AutoCAD2021.Base.Utilities.TestPlugin.cadwiki.AutoCAD2021.Base.Utilities.TestPlugin.UiRibbon
{
    internal class ButtonCreator
    {
        public static RibbonButton Create(string name, string text, string tooltip, Bitmap bitMap, string assemblyName, string fullClassName, string methodName, object[] parameters, AutoCADAppDomainDllReloader appDomainReloader, Assembly iExtensionAppAssembly)
        {


            var ribbonButton = new RibbonButton();
            ribbonButton.Name = name;
            ribbonButton.ShowText = true;
            ribbonButton.Text = text;
            ribbonButton.Size = RibbonItemSize.Standard;

            if (bitMap != null)
            {
                var image = FileStore.Bitmaps.CreateBitmapSourceFromGdiBitmapForAutoCADButtonIcon(bitMap);
                if (image != null)
                {
                    ribbonButton.Image = image;
                    ribbonButton.ShowImage = true;
                }
            }

            var uiRouter = new UiRouter(assemblyName, fullClassName, methodName, parameters, appDomainReloader, iExtensionAppAssembly);
            ribbonButton.CommandParameter = uiRouter;
            ribbonButton.CommandHandler = new GenericClickCommandHandler();
            ribbonButton.ToolTip = tooltip;
            return ribbonButton;
        }
    }
}